app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java

```diff
@@ -1,12 +1,29 @@
 package fr.neamar.kiss.dataprovider;
 
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
 
 import fr.neamar.kiss.loader.LoadSearchPojos;
 import fr.neamar.kiss.pojo.Pojo;
 import fr.neamar.kiss.pojo.SearchPojo;
 
 public class SearchProvider extends Provider<SearchPojo> {
+    private SharedPreferences prefs;
+    private static final Hashtable<String,String> searchProviderUrls = new Hashtable<>();
+
+    static {
+        searchProviderUrls.put("Google", "https://encrypted.google.com/search?q=");
+        searchProviderUrls.put("DuckDuckGo", "https://duckduckgo.com/?q=");
+        searchProviderUrls.put("Bing", "https://www.bing.com/search?q=");
+        searchProviderUrls.put("Yahoo", "http://search.yahoo.com/search?p=");
+        searchProviderUrls.put("Wikipedia", "https://en.wikipedia.org/wiki/");
+    }
 
```

Comment:
This get's sorted a little bit weird on my device (from bottom to top):

bing
duckduckgo
yahoo
google
wiki